controllers/articles.js

```diff
@@ -34,4 +40,118 @@ export default class ArticleController {
       return res.status(500).json({ success: false, error: [error.message] });
     }
   }
+
+  /**
+   * @description - Search for articles
+   * @static
+   * @param {Object} req - the request object
+   * @param {Object} res - the response object
+   * @memberof ArticleController
+   * @returns {Object} class instance
+   */
+  static async searchForArticles(req, res) {
+    try {
+      const searchTerms = ArticleController.generateSearchQuery(req.query);
+      const results = await Article.findAll({
+        raw: true,
+        where: {
+          ...searchTerms,
+        },
+        include: [{
+          model: User,
+          attributes: ['username', 'email', 'name', 'bio'],
+          as: 'author',
+        }]
+      });
+      return res.status(200).json({
+        results,
+        success: true
+      });
+    } catch (error) {
+      return res.status(500).json({
+        success: false,
+        message: 'Oops, something went wrong.'
+      });
+    }
+  }
+
+  /**
+   * @description - Generate queries for search and filter
+   * @static
+   * @param {Object}  searchTerms - the terms that the user wants to search for
+   * @memberof ArticleController
+   * @returns {Object} class instance
+   */
+  static generateSearchQuery(searchTerms) {
+    const {
+      author, term, endDate, startDate, tags, categoryId
+    } = searchTerms;
+
+    const filterFields = {
+      '$author.username$': {
+        $like: `%${author}%`
+      },
+      createdAt: {
+        $between: [startDate, endDate]
+      },
+      title: {
+        $like: `%${term}%`,
+      },
+      description: {
+        $like: `%${term}%`,
+      },
+      taglist: {
+        $contains: tags ? [...tags.split(',')] : []
+      },
+      categoryId: Number(categoryId),
+    };
+
+    if (!author) {
+      delete filterFields['$user.username$'];
+    }
+    if (!startDate || !endDate) {
+      delete filterFields.createdAt;
+    }
+    if (!categoryId) {
+      delete filterFields.categoryId;
+    }
+    return filterFields;
+  }
+
+  /**
+   * @description - Get article by slug
+   * @static
+   * @param {Object} req - the request object
+   * @param {Object} res - the response object
+   * @memberof ArticleController
+   * @returns {Object} class instance
+   */
+  static async getArticleBySlug(req, res) {
+    try {
+      const {
+        params: {
+          slug
+        }
+      } = req;
+      const article = await Article.findOne({
+        where: {
+          slug
+        },
+        include: [{
+          as: 'author',
+          model: User,
+          attributes: ['username', 'email', 'name', 'bio'],
+        }]
+      });
+      return res.status(200).json({
+        success: true,
+        article: article.dataValues
+      });
+    } catch (error) {
+      return res.status(404).json({
+        success: false,
+        errors: ['Article not found.'],
```

Comment:
This should be an error, not errors

Reply:
Our convention uses `errors` and it's an array. If I change it to `error`, we won't be consistent and when we're working on the front end, we won't know the exact one to use. Let's keep being consistent so it'll be easy for us. Thanks.